CHANGELOG.md

```diff
@@ -18,12 +18,19 @@ All notable changes to this project will be documented in this file.
 - CKKS: uniformized the arguments' position for all methods of the `Encoder` interface.
 - CKKS: renamed `Encoder.EncodeNTT/New` to `Encoder.Encode/New`.
 - CKKS: renamed `Scale` to `DefaultScale` in `Parameters` and `ParametersLiteral`.
-- CKKS: renamed `Encoder.EncodeDiagMatrixBSGSAtLvl` to `Encoder.EncodeDiagMatrixBSGS`.
+- CKKS: renamed `PtDiagMatrix` to `LinearTransform`
+- CKKS: changed constructors of `LinearTransform` structs from `Encoder.EncodeDiagMatrix*` to `NewLinearTransform*`.
 - CKKS: added `EncodeSlotsQP` to encode on `rlwe.PolyQP` and create custom `PtDiagMatrix`.
 - CKKS: `GetPrecisionStats` now also accepts `[]float64`.
 - CKKS: `Approximate` now accepts `func(float64)float64` and `func(complex128)complex128`
 - CKKS: approximation intervals are now `float64` instead of `complex128`.
+- CKKS: added `PolynomialVector` struct, which allows to evaluate multiple different polynomials on the same ciphertext with `Evaluator.EvaluatePoly`.
+- CKKS: `PtDiagMatrix` has been renamed `LinearTransform`
+- CKKS: encoding methods of `Encoder.Encode*` for diagonal matrix have been removed.
+- CKKS: added `AllocateLinearTransform`, `EncodeOnAllocatedLinearTransform`, `NewLinearTransform` and `NewLinearTransformBSGS`.
```

Comment:
If there is really a need for pre-alocating LinearTransform types, then we should follow the same logic as for `Plaintext` and `Ciphertext` types: `NewLinearTransform` is the allocator and `EncodeLinearTranform` initializes it with actual data.

Reply:
`NewLinearTranform` -> Allocates a `LinearTransform`
`EncodeLinearTransform` -> Encodes the actual data into a `LinearTransform`
`GenLinearTranform` -> (Si besoin) does the two calls

Reply:
Resolved